lib/nats.js

```diff
@@ -843,6 +865,9 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
  */
 
 Client.prototype.subscribe = function(subject, opts, callback) {
+  if (this.closed) {
+    this.emit('error', SUB_CONN_CLOSED_ERR);
```

Comment:
No, emitting an 'error' here will automatically throw an exception if it is not otherwise handled.  I think that is sufficient.

I spent some time looking over the read loop, and no, I don't see any good alternatives without a significant restructuring... and I don't have enough idea about the real performance characteristics of NodeJS to say whether that would be worth doing.  I know it doesn't fit with the usual layout, and synchronous loops are always dangerous in NodeJS, but I don't (yet?) have a good opinion on a reorganisation, there.  yieldTime works